### PR TITLE
GO-2249: add basic snapshot validation

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

Some users now use PB import to import their own snapshots (for example to test own imports)
Add snapshot validation to avoid errors during its creation. It also can be extensible 

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2249/panic-during-objectimport-in-latest-version
### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: core/block/import/pb/converter.go [hunk 1]
```diff
@@ -541,3 +541,7 @@ func (p *Pb) filterObjects(objectTypesToImport widgets.ImportWidgetFlags, object
 	}
 	return rootObjects
 }
+
+func (p *Pb) validateSnapshot(snapshot *pb.SnapshotWithType) bool {
```
Comment (KirillSto) [inline core/block/import/pb/converter.go:545]: Maybe rename it to `isSnapshotEmpty`?
1. More specific name for check
2. Starting with `is`, so we understand it returns boolean
3. We do not need to perform negation twice

**Review by KirillSto — CHANGES_REQUESTED** [review]

File: core/block/import/pb/converter.go (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/import/pb/converter.go:545]: I called it validateSnapshot, because we can add other rules in the future

Comment (AnastasiaShemyakinskaya) [reply core/block/import/pb/converter.go]: 1. Rename to isSnapshotValid 
2. `We do not need to perform negation twice`. What do you mean?

Comment (KirillSto) [reply core/block/import/pb/converter.go]: Inside isSnapshotValid we perform negation on result of disjunction. It is the first time
Second time we perform it on method's return value (!valid)

**Review by fat-fellow — APPROVED** [review]

File: core/block/import/pb/converter.go (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/import/pb/converter.go]: It is correct, I suppose
If disjunction is positive, it means, that snapshot is not valid. 
If snapshot is empty, the result of function is false - snapshot is not valid. if !valid, then we return error.
Or may be I missed something?

Comment (KirillSto) [reply core/block/import/pb/converter.go]: Yes, it is correct. Just redundant computing :slightly_smiling_face: 
If validation function will grow and will be reused somewhere else, it is ok to leave it as is

**Review by KirillSto — APPROVED** [review]

File: core/block/import/pb/converter.go (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/import/pb/converter.go]: I think it's more clear for function, which checks, if snapshot is valid. We can see it from the code, that if snapshot is not valid (!valid), then we return error.

**Pull request merged**